lib/cylc/task_pool.py

```diff
@@ -720,7 +720,7 @@ def reload_taskdefs(self):
         LOG.info("Reloading task definitions.")
         # Log tasks orphaned by a reload that were not in the task pool.
         for task in self.orphans:
-            if task not in [tsk.tdef.name for tsk in self.get_all_tasks()]:
+            if task not in (tsk.tdef.name for tsk in self.get_all_tasks()):
```

Comment:
The number of times `self.get_all_tasks()` is called is the same as the length of `self.orphans`? Would it be more efficient to pre-calculate the list of task names outside? Alternatively, would it be better to get the list of task name from `self.config.taskdefs`?